pkg/api/validation/validation.go

```diff
@@ -252,6 +252,10 @@ func ValidateObjectMetaUpdate(old, meta *api.ObjectMeta) errs.ValidationErrorLis
 	} else {
 		meta.CreationTimestamp = old.CreationTimestamp
 	}
+	// an object can never remove a deletion timestamp
+	if !old.DeletionTimestamp.IsZero() {
```

Comment:
We define IsZero for *util.Time so that we don't need the nil check (a nil util.Time is zero)

----- Original Message -----
> > @@ -252,6 +252,10 @@ func ValidateObjectMetaUpdate(old, meta
> > *api.ObjectMeta) errs.ValidationErrorLis
> >  	} else {
> >  		meta.CreationTimestamp = old.CreationTimestamp
> >  	}
> > +	// an object can never remove a deletion timestamp
> > +	if !old.DeletionTimestamp.IsZero() {
> 
> DeletionTimestamp can be nil.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/8641/files#r31464084